core/src/main/java/org/apache/iceberg/deletes/RoaringPositionBitmap.java

```diff
@@ -0,0 +1,309 @@
+   */
+  public boolean contains(long pos) {
+    validatePosition(pos);
+    int high = highBytes(pos);
+    int low = lowBytes(pos);
+    return high < bitmaps.length && bitmaps[high].contains(low);
+  }
+
+  /**
+   * Adds all positions from the other bitmap to this bitmap, modifying this bitmap in place.
+   *
+   * @param that the other bitmap
+   */
+  public void or(RoaringPositionBitmap that) {
+    allocateBitmapsIfNeeded(that.bitmaps.length);
+    for (int index = 0; index < that.bitmaps.length; index++) {
+      bitmaps[index].or(that.bitmaps[index]);
+    }
+  }
+
+  /**
+   * Indicates whether the bitmap is empty.
+   *
+   * @return true if the bitmap is empty, false otherwise
+   */
+  public boolean isEmpty() {
+    return cardinality() == 0;
+  }
+
+  /**
+   * Returns the number of positions in the bitmap.
+   *
+   * @return the number of set positions
+   */
+  public long cardinality() {
+    long cardinality = 0L;
+    for (RoaringBitmap bitmap : bitmaps) {
+      cardinality += bitmap.getLongCardinality();
+    }
+    return cardinality;
+  }
+
+  /**
+   * Run-length compresses the bitmap if it is more space efficient.
+   *
+   * @return whether the bitmap was compressed
+   */
+  public boolean runOptimize() {
+    boolean changed = false;
+    for (RoaringBitmap bitmap : bitmaps) {
+      changed |= bitmap.runOptimize();
+    }
+    return changed;
+  }
+
+  /**
+   * Iterates over all positions in the bitmap.
+   *
+   * @param consumer a consumer for positions
+   */
+  public void forEach(LongConsumer consumer) {
+    for (int index = 0; index < bitmaps.length; index++) {
+      forEach(bitmaps[index], index, consumer);
+    }
+  }
+
+  /**
+   * Returns the number of bytes required to serialize the bitmap.
+   *
+   * @return the serialized size in bytes
+   */
+  public long serializedSizeInBytes() {
+    long size = BITMAP_COUNT_SIZE_BYTES;
+    for (RoaringBitmap bitmap : bitmaps) {
+      size += BITMAP_KEY_SIZE_BYTES + bitmap.serializedSizeInBytes();
+    }
+    return size;
+  }
+
+  /**
+   * Serializes the bitmap using the portable serialization format described below.
+   *
+   * <ul>
+   *   <li>The number of 32-bit Roaring bitmaps, serialized as 8 bytes
+   *   <li>For each 32-bit Roaring bitmap, ordered by unsigned comparison of the 32-bit keys:
+   *       <ul>
+   *         <li>The key stored as 4 bytes
+   *         <li>Serialized 32-bit Roaring bitmap using the standard format
+   *       </ul>
+   * </ul>
+   *
+   * <p>Note the byte order of the buffer must be little-endian.
+   *
+   * @param buffer the buffer to write to
+   * @see <a href="https://github.com/RoaringBitmap/RoaringFormatSpe">Roaring bitmap spec</a>
+   */
+  public void serialize(ByteBuffer buffer) {
+    validateByteOrder(buffer);
+    buffer.putLong(bitmaps.length);
+    for (int index = 0; index < bitmaps.length; index++) {
+      buffer.putInt(index);
+      bitmaps[index].serialize(buffer);
```

Comment:
Is there a way to skip any that have no deletes?

Reply:
This would require 2 passes in `serialize`, as we first have to persist the number of bitmaps as the first element. I can make it work but not sure it is worth the complexity. Let me know.